File: 노영주/38주차/기지국설치.js
function solution(n, stations, w) {
  const COVERAGE_RANGE = 2 * w + 1;
  let answer = 0;
  let leftIdx = 1;

  stations.forEach(station => {
    const [left, right] = [station - w, station + w];
    if (left > leftIdx) {
      answer += Math.ceil((left - leftIdx) / COVERAGE_RANGE);
    }
    leftIdx = right + 1;
  });

  //최종 기지국 '설치 완료' 영역을 알아내야 한다.
  leftIdx--;

  if (leftIdx < n) {
    answer += Math.ceil((n - leftIdx) / COVERAGE_RANGE);
  }

  return answer;
}

File: 윤호선/38주차/기지국 설치.cpp
#include <iostream>
#include <vector>
using namespace std;

int solution(int n, vector<int> stations, int w)
{
    int answer = 0;
    int x=1;
    
    for(int i=0;i<stations.size();i++){
        int y=stations[i]-w;
        int dis=y-x;
        
        if(y<0){
            x=stations[i]+w+1;
            continue;
        }
    
        if(dis>0){ 
            answer+=dis/(2*w+1);
            if(dis%(2*w+1)!=0)++answer;
        }
        x=stations[i]+w+1;
    }
    
    if(x<=n){
        answer+=(n-x+1)/(2*w+1);
        if((n-x+1)%(2*w+1)!=0)++answer;
    }
   

    return answer;
}